### ISSUE TITLE
Error when `member` as a named type

### ISSUE THREAD

**Issue by pksunkara** [apiaryio/drafter#401]
That said, the following API Blueprint will raise an unintuitive error message, I would treat this as a bug in [Drafter](https://github.com/apiaryio/drafter) the underlying API Blueprint parser. We should improve the error message to make it clear to users that they are using reserved keywords incorrectly in cases we can detect.

```apib
## Data Structures
### member (object)
```

```
error: (4)  You can not register a basic element
```

However, if I use `member` as titlecase I don't get an error at all:

```apib
## Data Structures
### Member (object)
```

**Comment by w-vi** [comment]
Can we consider this as fixed by #405 ? The error message has been improved.

**Comment by pksunkara** [comment]
Yup.
